### PR TITLE
Isolate zimcheck testable code in separated file

### PR DESCRIPTION
**Pull request by MiguelRocha**
The current zimcheck class/file structure do not allow for an easy automated testing.

This PR moves the testable code from zimcheck main source file to zimfilecheks making it isolated and therefore testable.

It also adds several unit tests to zimfilechecks.

### CODE REVIEW

**Comment by kelson42** [comment]
@MiguelRocha I don’t get the purpose of this draft MR. We already have a zimcheck bin! Which ticket is getting fixed with it?

**Comment by MiguelRocha** [comment]
> @MiguelRocha I don’t get the purpose of this draft MR. We already have a zimcheck bin! Which ticket is getting fixed with it?

The current zimcheck class/file structure do not allow for an easy automated testing (its basically all code inside a single file). What this PR does is just separate the code within different files and makes it testable (with this PR the zimcheck code coverage will jump to 65%). Also, and IMO more importantly we can avoid having regressions by easily introducing unit testing when we have this kind of bugs: https://github.com/openzim/zim-tools/pull/148#issuecomment-674881789

(Sorry for not adding a description in the PR, I'll write a more clean one)

**Comment by kelson42** [comment]
@MiguelRocha Thx, makes perfectly sense to me now as well.

**Comment by kelson42** [comment]
@MiguelRocha Any chance we could get aPR ready to review for tomorrow?

**Comment by MiguelRocha** [comment]
@mgautierfr rebased in master and fixed commit history. It pushes code coverage of zim-tools to 31%. Please let me know your comments.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
This is nice but maybe not enough.
It would be nice to also split this too long `test_articles` into small functions and test them individually (without using a zim file).

But it is a good step and we can split/test more in another PR. I will not block this one for that.

(But we can move the code in subdirectory in this PR)

File: src/meson.build [hunk 1]
```diff
@@ -26,7 +26,7 @@ executable('zimsplit', 'zimsplit.cpp',
   dependencies: libzim_dep,
   install: true)
 
-executable('zimcheck', 'zimcheck.cpp', 'tools.cpp',
+executable('zimcheck', 'zimcheck.cpp', 'zimfilechecks.cpp', 'tools.cpp',
```
[the remark this answers is not in the document]
Comment (MiguelRocha) [reply src/meson.build]: Good idea. Done.

**Comment by MiguelRocha** [comment]
> This is nice but maybe not enough.
> It would be nice to also split this too long `test_articles` into small functions and test them individually (without using a zim file).
> 
> But it is a good step and we can split/test more in another PR. I will not block this one for that.
> 
> (But we can move the code in subdirectory in this PR)

Yes I agree we should split the huge function. Lets do it on other PR.

I've moved the code to a zimcheck directory. If its good I think we can merge.

**Comment by kelson42** [comment]
@MiguelRocha Can you please rebase (again) on `master`?

**Comment by MiguelRocha** [comment]
@kelson42 @mgautierfr rebased into master.

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**